core/app/models/spree/address.rb

```diff
@@ -11,23 +11,24 @@ class Address < Spree::Base
     belongs_to :country, class_name: "Spree::Country", optional: true
     belongs_to :state, class_name: "Spree::State", optional: true
 
-    validates :firstname, presence: true, if: -> { name.blank? || lastname? }
-    validates :name, presence: true, unless: :firstname?
     validates :address1, :city, :country_id, presence: true
     validates :zipcode, presence: true, if: :require_zipcode?
     validates :phone, presence: true, if: :require_phone?
 
+    validate :validate_name
     validate :state_validate
     validate :validate_state_matches_country
 
     alias_attribute :first_name, :firstname
     alias_attribute :last_name, :lastname
     alias_attribute :full_name, :name
 
+    DEPRECATION = { firstname: :name, lastname: :name, full_name: :name }
     DB_ONLY_ATTRS = %w(id updated_at created_at)
     TAXATION_ATTRS = %w(state_id country_id zipcode)
+    DEPRECATED_ATTRS = DEPRECATION.stringify_keys.keys
```

Comment:
Why `stringify_keys` is needed here?

Reply:
To be consistent with `SpreeAddress.value_attributes` and its callers, they work only with string hash keys https://github.com/solidusio/solidus/pull/3447/files/16fc264d3acc1ff743c8aac33b5cc93dd47bf737..93db0be1d74d5666145d6f0e7f6a052a98c06e28#diff-14b461f472ff6adbfd491ed5674e4e3aR65

Reply:
Isn't that method already calling `stringify_keys` on the first line?

```ruby
      base = base_attributes.stringify_keys.merge(merge_attributes.stringify_keys)
```

Reply:
Yes but it is used in `except` on string keyed hashes in in `self.factory` and `==`

Reply:
You are right, I thought it was passed to the method as an argument.